yarn-project/simulator/src/public/public_processor/public_processor.ts

```diff
@@ -332,12 +324,12 @@ export class PublicProcessor implements Traceable {
     // b) always had a txHandler with the same db passed to it as this.db, which updated the db in buildBaseRollupHints in this loop
     // To see how this ^ happens, move back to one shared db in test_context and run orchestrator_multi_public_functions.test.ts
     // The below is taken from buildBaseRollupHints:
-    await this.treesDB.appendLeaves(
+    await this.merkleTree.appendLeaves(
       MerkleTreeId.NOTE_HASH_TREE,
       padArrayEnd(processedTx.txEffect.noteHashes, Fr.ZERO, MAX_NOTE_HASHES_PER_TX),
     );
     try {
-      await this.treesDB.batchInsert(
+      await this.merkleTree.batchInsert(
```

Comment:
I had a version of this PR where I was using the PublicTreesDB for everything, but I figured it's probably more efficient to use batchInsert here, since it both inserts and pads.